### DISCUSSION TITLE
folder picker besides file picker

### DISCUSSION THREAD

**Discussion by da-i** [zauberzeug/nicegui#3793] [Q&A] [answered]
### Question

I've been looking for a solution to select a folder in native mode. But so far I've not been successful.

There was a good discussion on file&folder pickers, but that only resulted in a file picker:
https://github.com/zauberzeug/nicegui/discussions/283

I've tried below, and you can drag in a folder into the  ui.upload d&d box. But it trows an error when uploaded. 
And the create_file_dialog does exactly what it says on the tin: select a file. 

```py
async def choose_file():
    files = await app.native.main_window.create_file_dialog(allow_multiple=True)
    for file in files:
        ui.notify(file)


class AppUI:
    def __init__(self):
        app.native.settings['ALLOW_DOWNLOADS'] = True

        self.left_drawer = self.fill_left_drawer()
        self.right_drawer = self.fill_right_drawer()
        self.header = self.fill_header()
        self.footer = self.fill_footer()

        ui.label('CONTENT')
        ui.button('choose file', on_click=choose_file)
        x = ui.upload(on_upload=lambda e: ui.notify(f'Uploaded {e.name}'), auto_upload=True).classes('max-w-full')
```

TL;DR: Would it not be nice to be able to select a folder for native apps? this might be useful for outputs, or for example to process all images in a given folder

**Comment by falkoschindler** [comment, marked as answer]
Hi @da-i,

You can do that by passing a `dialog_type` to `create_file_dialog`:

```py
folders = await app.native.main_window.create_file_dialog(allow_multiple=True, dialog_type=webview.FOLDER_DIALOG)
```

**Reply by frankhuurman** [reply]
Tried to use this on Arch Linux and it resulted in:
```
Traceback (most recent call last):
  File "/usr/lib/python3.13/multiprocessing/queues.py", line 262, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/usr/lib/python3.13/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^
_pickle.PicklingError: Can't pickle <function FOLDER_DIALOG at 0x7f1836037420>: it's not the same object as webview.FOLDER_DIALOG
```

Seems that webview.FOLDER_DIALOG is deprecated, but **webview.FileDialog.FOLDER** makes it work again:
`files = await app.native.main_window.create_file_dialog(allow_multiple=True, dialog_type=webview.FileDialog.FOLDER)`

I was afraid I had to fallback to the good ol' ugly tkinter filedialog.askdirectory() but I'm happy that I don't have to!
Just wish Quasar made a QFolder picker or provide this option to a QFile component.
